### PR TITLE
Store "use ssh" setting in local storage

### PR DESCRIPTION
**Pull request by jpbernius**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [X] I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage)
- [X] ~I added `authorities` to all new routes and check the course groups for displaying navigation elements (links, buttons)~
- [X] I documented the TypeScript code using JSDoc style.
- [X] ~I added multiple screenshots/screencasts of my UI changes~
- [X] ~I translated all the newly inserted strings into German and English~

### Motivation and Context
#2317 and #2363
TL;DR: Remember if `git clone ssh://` or `git clone https://`

### Description
Use the `localStorage` to remember which setting was last ued.

### Steps for Testing

1. Find programming exercise (and start if needed)
2. Click on "Clone exercise"
3. Find `https://` url.
4. Click "USE SSH" and find `ssh://` url.
5. Reload page, repeat 1 & 2.
6. Find `ssh://` url.
7. Click "USE HTTPS" and find `https://` url.
8. Reload page, repeat 3-7 until it gets boring.

### Test Coverage
* exercies-details-tsudent-actions.component.ts 71% -> 75%

### CODE REVIEW

**Review by simonzilker — APPROVED** [review]
Works perfectly!

**Review by airelawaleria — APPROVED** [review]
tested on ts1. works fine

**Review by JohannesWeiss — APPROVED** [review]

**Review by marvinchimeka — APPROVED** [review]

**Review by EneaGore — APPROVED** [review]

**Review by JannisBittlmayer — APPROVED** [review]
Tested on TS3, works great. Only thing I noticed is that this doesn't save when logging out and in again, but I assume this is intended

**Review by krusche — APPROVED** [review]
